### ISSUE TITLE
provide one "partial" field per entry in aggregated responses

### ISSUE THREAD

**Issue by severo** [huggingface/datasets-server#1532]
For example, https://datasets-server.huggingface.co/size?dataset=c4 only provides a global `partial: true` field and the response does not explicit that the "train" split is partial, while the "test" one is complete.

Instead, every entry in `configs` and `splits` should include its own `partial` field, to be able to show this information in the viewer (selects)

- currently:
  <img width="1528" alt="Capture d’écran 2023-07-19 à 16 00 28" src="https://github.com/huggingface/datasets-server/assets/1676121/92d27982-0fa3-44f2-a73f-a0ae614da40c">
- ideally, something like:
  <img width="1529" alt="Capture d’écran 2023-07-19 à 16 01 39" src="https://github.com/huggingface/datasets-server/assets/1676121/c638af93-30de-4ab7-8fdd-389202d41c88">

**Comment by severo** [comment]
Note that this means changing the format (and implementation) of the config-parquet-and-info step, and recomputing all its artifacts 😬

Also: the field `partial` should be added to every entry of `splits` in the /info response (or provided in another format, if we want to preserve the "info" as exported by the datasets library)

**Comment by severo** [comment]
Maybe https://github.com/huggingface/moon-landing/pull/7079 is sufficient for now.

**Comment by severo** [comment]
> Note that this means changing the format (and implementation) of the config-parquet-and-info step, and recomputing all its artifacts 😬

I think we should store which splits are partial and which are complete. Opening an issue for that -> https://github.com/huggingface/dataset-viewer/issues/2809, and this one will depend on it.

**Comment by lhoestq** [comment]
Note that we can get this info per split already for free for most datasets:
- if there is only one split then its `partial` value is equal to the `partial` value at config level
- if there is the split size info in the YAML in the README.md then the `partial` value is True if it matches the size generated by config-parquet-and-info

So actually we should be able to retrieve most of the `partial` values no ?

**Comment by severo** [comment]
yes, it would be a good way to migrate the cache entries to the new schema instead of recomputing in #2809
